changes/1817.feature.md

```diff
@@ -0,0 +1 @@
+Improve Postgres connection management by retrying transactions instead of connections and add `pool-recycle` config to replace timeout connections.
```

Comment:
```suggestion
Improve Postgres connection management by retrying transactions inside connections instead of retrying entire connections, and add `pool-recycle` config to drop and replace timed-out connections.
```